src/Okta.Sdk/OktaClient.cs

```diff
@@ -3,59 +3,69 @@
+            var compiled = CompileFromConfigurationSources(apiClientConfiguration);
+            var config = new OktaClientConfiguration();
+            compiled.GetSection("okta").GetSection("client").Bind(config);
 
-            //var homeOktaJsonLocation = HomePath.Resolve("~", ".okta", "okta.json");
-            //var homeOktaYamlLocation = HomePath.Resolve("~", ".okta", "okta.yaml");
+            ThrowIfInvalidConfiguration(config);
 
-            //var applicationAppSettingsLocation = Path.Combine(configurationFileRoot ?? string.Empty, "appsettings.json");
-            //var applicationOktaJsonLocation = Path.Combine(configurationFileRoot ?? string.Empty, "okta.json");
-            //var applicationOktaYamlLocation = Path.Combine(configurationFileRoot ?? string.Empty, "okta.yaml");
+            var realLogger = logger ?? NullLogger.Instance;
 
-            //var configBuilder = new ConfigurationBuilder()
-            //    .AddYamlFile(homeOktaYamlLocation, optional: true, root: "okta")
-            //    .AddJsonFile(homeOktaJsonLocation, optional: true, root: "okta")
-            //    .AddJsonFile(applicationAppSettingsLocation, optional: true)
-            //    .AddYamlFile(applicationOktaYamlLocation, optional: true, root: "okta")
-            //    .AddJsonFile(applicationOktaJsonLocation, optional: true, root: "okta")
-            //    .AddEnvironmentVariables("okta", "_", root: "okta")
-            //    .AddObject(apiClientConfiguration, root: "okta");
+            var requestExecutor = new DefaultRequestExecutor(config, realLogger);
+            DataStore = new DefaultDataStore(requestExecutor, new DefaultSerializer(), realLogger);
+        }
 
-            //var config = configBuilder.Build();
+        private static FlexibleConfiguration.Abstractions.IConfigurationRoot CompileFromConfigurationSources(OktaClientConfiguration apiClientConfiguration = null)
+        {
+            string configurationFileRoot = Directory.GetCurrentDirectory();
 
-            // TODO: validate configuration
+            var homeOktaYamlLocation = HomePath.Resolve("~", ".okta", "okta.yaml");
 
-            //_configuration = apiClientConfiguration;
+            var applicationAppSettingsLocation = Path.Combine(configurationFileRoot ?? string.Empty, "appsettings.json");
+            var applicationOktaYamlLocation = Path.Combine(configurationFileRoot ?? string.Empty, "okta.yaml");
 
-            //_logger = logger ?? NullLogger.Instance;
+            var configBuilder = new ConfigurationBuilder()
+                .AddYamlFile(homeOktaYamlLocation, optional: true)
+                .AddJsonFile(applicationAppSettingsLocation, optional: true)
+                .AddYamlFile(applicationOktaYamlLocation, optional: true)
+                .AddEnvironmentVariables("okta", "_", root: "okta")
+                .AddObject(apiClientConfiguration, root: "okta:client");
```

Comment:
Just to make sure I understand what is going on here: Every appended method will override the existing config, right? So if we have two `okta.yaml` files, one in the root dir and the second in our project path, the `applicationOktaYamlLocation` will be the source of truth.

Is this correct?